File: 이건희/level2/문자열 압축.py
def solution(s):
    result = len(s)
    len_s = len(s)
    
    for i in range(1, len(s) // 2 + 1):
        box = [s[j:j+i] for j in range(0, len(s), i)]
        len_s = len(s)
        cnt = 1

        for j in range(len(box)):
            if j == len(box) - 1:
                if cnt > 1:
                    len_s -= i * cnt
                    len_s += i + len(str(cnt))
            else:
                if box[j] == box[j + 1]:
                    cnt += 1
                elif cnt > 1:
                    len_s -= i * cnt
                    len_s += i + len(str(cnt))
                    cnt = 1
        
        result = min(result, len_s)
        
    return result

File: 이훈일/Programmers/Level2/문자열 압축.java
class Solution {
    public static int solution(String s) {
        int result = Integer.MAX_VALUE;
        for(int i = 1; i <= s.length(); i++) {
            result = Math.min(compress(s, i), result);
        }
        
        return result;
    }
    
    public static int compress(String s, int len) {
        String pre = s.substring(0, len);
        boolean isSame = false;
        int totalCnt = 0;
        int sameCnt = 0;
        for(int i = len; i < s.length(); i += len) {
            String cur = "";
            if(i+len > s.length()) {
                cur = s.substring(i);
            } else {
                cur = s.substring(i, i+len);
            }
            
            if(!pre.equals(cur)) {
                if(isSame) {
                    totalCnt += len + String.valueOf(sameCnt).length();
                } else {
                    totalCnt += len;
                }
                isSame = false;
            } else {
                if(isSame) {
                    sameCnt++;
                } else {
                    isSame = true;
                    sameCnt = 2;
                }
            }
            pre = cur;
        }
        
        if(isSame) {
            if(sameCnt > 9) {
                totalCnt += len + String.valueOf(sameCnt).length();
            } else {
                totalCnt += len + 1;
            }
        } else {
            totalCnt += pre.length();
        }
        
        return totalCnt;
    }
}
